posts/2024-11-01-gsoc-2024-results.md

```diff
@@ -0,0 +1,119 @@
+
+Great work, Max!
+
+### Implementation of a Faster Register Allocator For Cranelift
+
+- Contributor: [Demilade Sonuga](https://github.com/d-sonuga)
+- Mentors: [Chris Fallin](https://github.com/cfallin) and [Amanieu d'Antras](https://github.com/Amanieu)
+- [Final report](https://d-sonuga.netlify.app/gsoc/regalloc-iii/)
+
+TODO
+
+### Improve Rust benchmark suite
+
+- Contributor: [Eitaro Kubotera](https://github.com/s7tya)
+- Mentor: [Jakub Beránek](https://github.com/kobzol)
+- [Final report](https://gist.github.com/s7tya/96dc1cae4ca9e59841a95f0f48d023d6)
+
+This project was relatively loosely defined, with the overarching goal of improving the user interface of the [Rust compiler benchmark suite](https://github.com/rust-lang/rustc-perf). Eitaro tackled this challenge from various angles at once. He improved the visualization of runtime benchmarks, which were previously a second-class citizen in the benchmark suite, by adding them to our [dashboard](https://perf.rust-lang.org/dashboard.html) and by implementing [historical charts](https://github.com/rust-lang/rustc-perf/pull/1922) of runtime benchmark results, which help us figure out how is a given benchmark behaving over a longer time span.
+
+Another improvement that he has worked on was embedding a profiler trace visualizer directly within the `rustc-perf` website. This was a challenging task, which required him to evaluate several visualizers and figure out a way how to include them within the source code of the benchmark suite in a non-disruptive way. In the end, he managed to integrate [Perfetto](https://ui.perfetto.dev/) within the suite website, and also performed various [optimizations](https://github.com/rust-lang/rustc-perf/pull/1968) to improve the performance of loading compilation profiles.
+
+Last, but not least, Eitaro also created a completely new user interface for the benchmark suite, which runs entirely in the [terminal](https://github.com/rust-lang/rustc-perf/pull/1955). Using this interface, Rust compiler contributors can examine the performance of the compiler without having to start the rustc-perf website, which can be challenging to deploy locally.
+
+Apart from the mentioned contributions, Eitaro also made a lot of other smaller improvements to various parts of the benchmark suite. Thank you for all your work!
+
+### Move cargo shell completions to Rust
+
+- Contributor: [shanmu](https://github.com/shannmu)
+- Mentor: [Ed Page](https://github.com/epage)
+- [Final report](https://hackmd.io/@PthRWaPvSmS_2Yu_GLbGpg/Hk-ficKpC)
+
+TODO
+
+### Rewriting Esoteric, Error-Prone Makefile Tests Using Robust Rust Features
+
+- Contributor: [Julien Robert](https://github.com/Oneirical)
+- Mentor: [Jieyou Xu](https://github.com/jieyouxu)
+- [Final report](https://oneirical.github.io/gsocfinal/)
+
+TODO
```

Comment:
The [`run-make` test suite][run-make] previously had tests based on `Makefile`s. However, this setup [posed two primary problems][initial-run-make-issue]:

1. You could only run the test suite under `windows-gnu` (e.g. MSYS2) and not natively under `windows-msvc`. This is problematic because `x86_64-pc-windows-msvc` is also a Tier 1 target, yet it doesn't receive native test coverage from the `run-make` test suite. Contributors also had to perform additional setup like MSYS2 to be able to run the test suite at all.
2. Some of the `Makefile` syntax are quite esoteric, with errors easy to silently ignore and mistakes very difficult to spot, even to multiple reviewers.

Julien helped to convert the `Makefile`-based `run-make` tests into plain Rust-based tests, supported by a test support library called `run_make_support`. It was not a trivial "rewrite this in Rust" kind of deal. In this project, Julien:

- Significantly improved the test documentation;
- Fixed multiple bugs that were present in the `Makefile` versions that had gone unnoticed for *years* -- some tests were never testing anything or silently ignored failures, so even if the subject being tested regressed, these tests would not have caught that.
- Added to and improved the test support library API and implementation; and
- Improved code organization within the tests to make them easier to understand and maintain.

As a result, I believe Julien has significantly improved the robustness of the `run-make` test suite, and improved the ergonomics of modifying existing `run-make` tests and authoring new `run-make` tests. Multiple contributors have expressed that they were more willing to work with the Rust-based `run-make` tests over the `Makefile` versions.

[run-make]: https://github.com/rust-lang/rust/tree/master/tests/run-make
[initial-run-make-issue]: https://github.com/rust-lang/rust/issues/40713

Reply:
(Feel free to compress this to edit as suitable)

Reply:
Thanks a lot! I tried to add some initial context, just to let people know what is the run-make test suite, and also did some editing. Let me know what do you think.

Reply:
Sounds good to me, thanks!